sdk/mongocluster/arm-mongocluster/src/models/models.ts

```diff
@@ -89,74 +74,313 @@ export function proxyResourceSerializer(item: ProxyResource) {
+      ? item.replicaParameters
+      : mongoClusterReplicaParametersSerializer(item.replicaParameters),
+    administrator: !item.administrator
+      ? item.administrator
+      : administratorPropertiesSerializer(item.administrator),
+    serverVersion: item["serverVersion"],
+    publicNetworkAccess: item["publicNetworkAccess"],
+    highAvailability: !item.highAvailability
+      ? item.highAvailability
+      : highAvailabilityPropertiesSerializer(item.highAvailability),
+    storage: !item.storage ? item.storage : storagePropertiesSerializer(item.storage),
+    sharding: !item.sharding ? item.sharding : shardingPropertiesSerializer(item.sharding),
+    compute: !item.compute ? item.compute : computePropertiesSerializer(item.compute),
+    backup: !item.backup ? item.backup : backupPropertiesSerializer(item.backup),
+    previewFeatures: item["previewFeatures"],
+  };
 }
 
-/** The error detail. */
-export interface ErrorDetail {
-  /** The error code. */
-  readonly code?: string;
-  /** The error message. */
-  readonly message?: string;
-  /** The error target. */
-  readonly target?: string;
-  /** The error details. */
-  readonly details?: ErrorDetail[];
-  /** The error additional info. */
-  readonly additionalInfo?: ErrorAdditionalInfo[];
+/** Known values of {@link CreateMode} that the service accepts. */
+export enum KnownCreateMode {
+  /** Default */
+  Default = "Default",
+  /** PointInTimeRestore */
+  PointInTimeRestore = "PointInTimeRestore",
+  /** GeoReplica */
+  GeoReplica = "GeoReplica",
+  /** Replica */
+  Replica = "Replica",
 }
 
-/** The resource management error additional info. */
-export interface ErrorAdditionalInfo {
-  /** The additional info type. */
-  readonly type?: string;
-  /** The additional info. */
-  readonly info?: Record<string, any>;
+/**
+ * The mode that the Mongo Cluster is created with. \
+ * {@link KnownCreateMode} can be used interchangeably with CreateMode,
+ *  this enum contains the known values that the service supports.
+ * ### Known values supported by the service
+ * **Default** \
+ * **PointInTimeRestore** \
+ * **GeoReplica** \
+ * **Replica**
+ */
+export type CreateMode = string;
+
+/** Parameters used for restore operations */
+export interface MongoClusterRestoreParameters {
+  /** UTC point in time to restore a mongo cluster */
+  pointInTimeUTC?: Date;
+  /** Resource ID to locate the source cluster to restore */
+  sourceResourceId?: string;
 }
 
-/** The response of a PrivateEndpointConnectionResource list operation. */
-export interface _PrivateEndpointConnectionResourceListResult {
-  /** The PrivateEndpointConnectionResource items on this page */
-  value: PrivateEndpointConnectionResource[];
-  /** The link to the next page of items */
-  nextLink?: string;
+export function mongoClusterRestoreParametersSerializer(
+  item: MongoClusterRestoreParameters,
+): Record<string, unknown> {
+  return {
+    pointInTimeUTC: item["pointInTimeUTC"]?.toISOString(),
```

Comment:
why this is iso string? pls share the tsp link with me

Reply:
https://github.com/kazrael2119/azure-rest-api-specs/blob/c3474b5f97987003f73a08765b3450f2829eb6ea/specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp#L197-L198